copy-to-clipboard/index.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+let templateHtml = require('./copy-to-clipboard.html');
+let uniqueId = require('lodash.uniqueid');
+
+let component = {};
+
+component.restrict = 'E';
+component.template = templateHtml;
+component.controllerAs = 'vm';
+component.bindToController = true;
+component.bindings = {
+    copyText: '<',
+    buttonLabel: '@',
+    buttonClasses: '@?'
+};
+
+//@ngInject
+component.controller = function BdCopyToClipboard($log) {
+    let vm = this;
+
+    function clearSelection() {
+        let selection = window.getSelection ? window.getSelection() : document.selection;
+        if (selection) {
+            if (selection.removeAllRanges && typeof selection.removeAllRanges === 'function') {
+                selection.removeAllRanges();
+            } else if (selection.empty && typeof selection.empty === 'function') {
+                selection.empty();
+            }
+        }
+    }
+
+    function selectText(containerid) {
+        let range;
+        clearSelection();
+        if (document.selection) {
+            range = document.body.createTextRange();
+            range.moveToElementText(document.getElementById(containerid));
+            range.select();
+        } else if (window.getSelection) {
+            range = document.createRange();
+            range.selectNode(document.getElementById(containerid));
+            window.getSelection().addRange(range);
+        }
+    }
+
+    vm.$onInit = function onInit() {
+        vm.clipBoardTextId = uniqueId('clipboard-text-');
+    };
+
+    vm.copyTextToClipboard = function copyTextToClipboard() {
+        let wasCopySuccessful;
+        selectText(vm.clipBoardTextId);
+        try {
+            wasCopySuccessful = document.execCommand('copy');
+            if (wasCopySuccessful) {
+                $log.info('Copied to clipboard');
+            } else {
+                $log.error('Oops, this is not supported in your browser');
+            }
+        } catch (err) {
+            $log.error('Oops, this is not supported in your browser');
+        }
+        clearSelection();
+    };
+};
+
+let moduleName = 'buildium.angular-ui.copytoclipboard';
+
+angular.module(moduleName, [])
+.component('bdCopyToClipboard', component)
+
+module.exports = moduleName;
```

Comment:
github diff warns that there's no newline at end of file

we haven't added any linting to the project yet though so 🤷‍♂️